disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/exporters/MetricDiskExporter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.exporters;
+
+import io.opentelemetry.contrib.disk.buffering.internal.exporters.AbstractDiskExporter;
+import io.opentelemetry.contrib.disk.buffering.internal.serialization.serializers.SignalSerializer;
+import io.opentelemetry.contrib.disk.buffering.storage.StorageConfiguration;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ * This is a {@link MetricExporter} wrapper that takes care of intercepting all the signals sent out
+ * to be exported, tries to store them in the disk in order to export them later.
+ *
+ * <p>In order to use it, you need to wrap your own {@link MetricExporter} with a new instance of
+ * this one, which will be the one you need to register in your {@link MetricReader}.
+ */
+public final class MetricDiskExporter extends AbstractDiskExporter<MetricData>
+    implements MetricExporter {
+  private final MetricExporter wrapped;
+
+  /**
+   * @param wrapped - Your own exporter.
+   * @param rootDir - The directory to create this signal's cache dir where all the data will be
+   *     written into.
+   * @param configuration - How you want to manage the storage process.
+   */
+  public MetricDiskExporter(
+      MetricExporter wrapped, File rootDir, StorageConfiguration configuration) {
+    super(rootDir, configuration);
+    this.wrapped = wrapped;
+  }
+
+  @Override
+  protected String getStorageFolderName() {
+    return "metrics";
+  }
+
+  @Override
+  protected CompletableResultCode doExport(Collection<MetricData> metricData) {
+    return wrapped.export(metricData);
+  }
+
+  @Override
+  protected SignalSerializer<MetricData> getSerializer() {
+    return SignalSerializer.ofMetrics();
+  }
+
+  @Override
+  public CompletableResultCode export(Collection<MetricData> metrics) {
+    return onExport(metrics);
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    return wrapped.flush();
```

Comment:
why not just return success?
It's weird that flush should be propagated, but write gets buffered.

Reply:
Do you mean returning success without calling `wrapped.flush()`? Or just returning success afterward?

Reply:
without calling wrapped - the point of the disk exporter is to do all interactions with the wrapped exporter later.

Reply:
I see, my only concern is that some exporters tend to batch signals and send them later, so the same might happen even when the disk exporter delegates data to them, so in those cases this might cause that data to never get exported. However, I believe for that case it might be better to call `wrapped.flush()` right after the disk exporter delegates the data (during the reading process) to avoid these kinds of issues, and after that, we can simply remove the flush call from here and just return success right away as you suggest 👍  I'm glad I got this insight because without calling `flush` during the reading/delegating process, the disk exporter might be getting unnecessary delays during the delegation calls right now, so it's a great catch!

Reply:
Thank you, I've made the changes. Also, after double-checking, it seems like no exporters are batching anything (I must have confused them with some processors) and the ones available right now are always returning success here, so there's no need to do the flushing after the reading/delegating process either.